File: solutions/go/anagram/1/anagram.go
package anagram

import (
	"sort"
	"strings"
)

func SortString(word string) string {
	wordList := strings.Split(word, "")
	sort.Strings(wordList)
	return strings.Join(wordList, "")
}

func Detect(subject string, candidates []string) (result []string) {
	lowercaseSubject := strings.ToLower(subject)
	sortedSubject := SortString(strings.ToLower(subject))
	for _, candidate := range candidates {
		lowercaseCandidate := strings.ToLower(candidate)
		if lowercaseSubject != lowercaseCandidate && SortString(lowercaseCandidate) == sortedSubject {
			result = append(result, candidate)
		}
	}
	return
}

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    anagram_list = []
    word_list = word.split()
    word_list.sort()
    for candidate in candidates:
        candidate_list = candidate.split()
        candidate_list.sort()
        if word_list == candidate_list:
            anagram_list.append(candidate)
    return anagram_list
